src/Exchange/SwapExchange.php

```diff
@@ -18,28 +18,38 @@
 final class SwapExchange implements Exchange
 {
     /**
-     * @var SwapInterface
+     * @var Swap
      */
     private $swap;
 
     /**
-     * @param SwapInterface $swap
+     * @param Swap|ExchangeRateProvider $exchange
```

Comment:
If it's `SwapExchange` that uses `Swap`, shouldn't we make `ExchangerExchange` that uses `ExchangeRateProvider`?